scala_proto/private/proto_to_scala_src.bzl

```diff
@@ -0,0 +1,49 @@
+load(
+    "//scala/private:common.bzl",
+    "write_manifest_file",
+)
+load("//scala/private:rule_impls.bzl", "compile_scala")
+load("//scala_proto/private:dep_sets.bzl", "SCALAPB_DEPS", "GRPC_DEPS")
+
+
+def _root_path(f):
+    if f.is_source:
+        return f.owner.workspace_root
+    return "/".join([f.root.path, f.owner.workspace_root])
+
+
+def _colon_paths(data):
+    return ":".join([
+        f.path
+        for f in sorted(data)
+    ])
+
+
+def proto_to_scala_src(ctx, label, code_generator, compile_proto, include_proto, transitive_proto_paths, flags, jar_output):
+    worker_content = "{output}\n{included_proto}\n{flags_arg}\n{transitive_proto_paths}\n{inputs}\n{protoc}".format(
+        output = jar_output.path,
+        included_proto = "-" + ":".join(sorted(["%s,%s" % (f.root.path, f.path) for f in include_proto])),
+        # Command line args to worker cannot be empty so using padding
+        flags_arg = "-" + ",".join(flags),
+        transitive_proto_paths = "-" + ":".join(sorted(transitive_proto_paths)),
+        # Command line args to worker cannot be empty so using padding
+        # Pass inputs seprately because they doesn't always match to imports (ie blacklisted protos are excluded)
+        inputs = _colon_paths(compile_proto),
+        protoc = ctx.executable._protoc.path
+    )
+    argfile = ctx.actions.declare_file(
+        "%s_worker_input" % label.name,
+        sibling = jar_output,
+    )
+    ctx.actions.write(output = argfile, content = worker_content)
+    ctx.actions.run(
+        executable = code_generator.files_to_run,
```

Comment:
@johnynek even after registering the default toolchain in my WORKSPACE, I'm getting the following error:
```
ERROR: /private/var/tmp/_bazel_feri/27c97e79ab5b7e8bcdeb0d679a4bc60c/external/com_google_protobuf/BUILD:248:2: in @io_bazel_rules_scala//scala_proto/private:scalapb_aspect.bzl%scalapb_aspect aspect on proto_library rule @com_google_protobuf//:any_proto:
Traceback (most recent call last):
	File "/private/var/tmp/_bazel_feri/27c97e79ab5b7e8bcdeb0d679a4bc60c/external/com_google_protobuf/BUILD", line 248
		@io_bazel_rules_scala//scala_proto/private:scalapb_aspect.bzl%scalapb_aspect(...)
	File "/private/var/tmp/_bazel_feri/27c97e79ab5b7e8bcdeb0d679a4bc60c/external/io_bazel_rules_scala/scala_proto/private/scalapb_aspect.bzl", line 175, in _scalapb_aspect_impl
		proto_to_scala_src(ctx, target.label, code_generator, com..., <4 more arguments>)
	File "/private/var/tmp/_bazel_feri/27c97e79ab5b7e8bcdeb0d679a4bc60c/external/io_bazel_rules_scala/scala_proto/private/proto_to_scala_src.bzl", line 39, in proto_to_scala_src
		ctx.actions.run(executable = code_generator.file..., <6 more arguments>)
expected value of type 'File or string' for parameter 'executable', in method call run(FilesToRunProvider executable, list inputs, list outputs, string mnemonic, string progress_message, dict execution_requirements, list arguments) of 'actions'
```

Reply:
@gseitz FYI

Reply:
@tferi-da This needs a fix from a bazel update -- the latest rules require bazel 0.23 or above now i'm afraid

Reply:
@ianoc thanks for the tip!